File: cash.c
#include <stdio.h>
#include <cs50.h>
#include <unistd.h>
#include <math.h>
#include <string.h>

float cash;

int main(void)
{
    
    // Get positive float from user.
    do
    {
        cash = get_float("Change owed: ");
    }
    while (cash < 0);
    
    // Translate cash into cents.
    int cents = round(cash * 100);
    
    // Use C to count total coins.
    int C = 0;

    // Determine if quarters apply.
    if (cents / 25 >= 1)
    {
        // printf("Quarters available.\n");
        do
        {
            C++;
            cents -= 25;
        }
        while (cents / 25 >= 1);
        // printf("Cents left after quarters: %i\n", cents);
    }
    
    // Determine if dimes apply.
    if (cents / 10 >= 1)
    {
        // printf("Dimes available\n");
        do
        {
            C++;
            cents -= 10;
        }
        while (cents / 10 >= 1);
        // printf("Cents left after dimes: %i\n", cents);
    }
    
    // Determine if nickels apply.
    if (cents / 5 >= 1)
    {
        // printf("Nickels available.\n");
        do
        {
            C++;
            cents -= 5;
        }
        while (cents / 5 >= 1);
        // printf("Cents left after nickels: %i\n", cents);
    }

    // Determine if pennies apply.
    if (cents > 0)
    {
        // printf("Pennies available.\n");
        do
        {
            C++;
            cents--;
        }
        while (cents > 0);
    }

    // printf("Coins to give: %i\n", C);
    printf("%i\n", C);

}

File: pset6/cash/cash.py
from cs50 import get_float

while True:
    change_float = get_float("Change owed: ")
    if change_float > 0:
        break

change = change_float * 100
total_coins = 0
# quarters = 0
# dimes = 0
# nickels = 0
# pennies = 0

while change >= 25:
    change -= 25
    # quarters += 1
    total_coins += 1
    
while change >= 10:
    change -= 10
    # dimes += 1
    total_coins += 1
    
while change >= 5:
    change -= 5
    # nickels += 1
    total_coins += 1

while change >= 1:
    change -= 1
    # pennies += 1
    total_coins += 1
    
print(f"{total_coins}")
# print(f"Quarters: {quarters}")
# print(f"Dimes: {dimes}")
# print(f"Nickels: {nickels}")
# print(f"Pennies: {pennies}")